taskcluster/translations_taskgraph/transforms/continuation.py

```diff
@@ -0,0 +1,339 @@
+        # "corpus-align-distillation" is the stage, while "{src_locale}-{trg_locale}" is the
+        # actual job name at this point in time.
+        stage = job.get("attributes", {}).get("stage", "")
+        label = f"{stage}-{job['name']}"
+
+        # Ensure continuation tasks don't get produced unless they are explicitly requested.
+        if stage == "continuation-corpus":
+            if (
+                not corpus_backtranslations
+                and label == "continuation-corpus-backtranslations-{src_locale}-{trg_locale}"
+            ):
+                continue
+            if (
+                not corpus_parallel
+                and label == "continuation-corpus-parallel-{src_locale}-{trg_locale}"
+            ):
+                continue
+            if (
+                not corpus_distillation
+                and label == "continuation-corpus-distillation-{src_locale}-{trg_locale}"
+            ):
+                continue
+        if not vocab and stage == "continuation-vocab":
+            continue
+        if (
+            not model_backwards
+            and stage == "continuation-model"
+            and label == "continuation-backwards-{src_locale}-{trg_locale}"
+        ):
+            continue
+
+        if corpus_parallel:
+            if stage == "corpus-merge-parallel":
+                # Skip any jobs that should never be produced:
+                continue
+
+            rewrite_dependencies(
+                job,
+                old_task="corpus-merge-parallel",
+                new_task="continuation-corpus-parallel",
+            )
+            if corpus_parallel.get("alignments"):
+                rewrite_dependencies(
+                    job,
+                    old_task="corpus-align-parallel",
+                    new_task="continuation-corpus-parallel",
+                )
+
+        if corpus_backtranslations:
+            if stage in {"corpus-merge-mono", "evaluate-backwards"}:
+                # Skip any jobs that should never be produced:
+                continue
+
+            rewrite_dependencies(
+                job,
+                old_task="corpus-merge-mono-trg",
+                new_task="continuation-corpus-backtranslations",
+            )
+            rewrite_dependencies(
+                job,
+                old_task="corpus-merge-mono-src",
+                new_task="continuation-corpus-backtranslations",
+            )
+
+            if corpus_backtranslations.get("alignments"):
+                if stage == "corpus-align-backtranslations":
+                    continue
+                rewrite_dependencies(
+                    job,
+                    old_task="corpus-align-backtranslations",
+                    new_task="continuation-corpus-backtranslations",
+                )
+
+        if corpus_distillation:
+            if stage in {
+                # "distillation-parallel-src-extract-best",
+                "train-teacher-model",
```

Comment:
What if we want to add another step to the pipeline? It will be tricky to find these lists. Would moving them to the top of the file help?

Reply:
This was more of a mitigation to ensure I got it all correct, so it should be fine to add new stages. I'm updating the comment here:

```
                # Skip any jobs that should never be produced. This helps ensure
                # that if they do somehow get produced, the taskgraph will fail to
                # fully resolve.
```